### PR TITLE
fix(forms): updateOn should check if change occurred

### LINKED ISSUE #20259: updateOn: 'blur' is firing an ngModelChange, regardless of the ngModel value changing
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
I'm using the new ngModelOptions updateOn to reduce the ngModelChange event firing from every key press to when the user blurs. However, the ngModelChange is now firing regardless of whether the ngModel has changed or not.

So the user can tab into the input, make no changes, tab out and the ngModelChange event is fired.

## Expected behavior
The ngModelChange should only fire when the ngModel actually changes. As in, user tabs/selects the input makes one or several change(s) then tabs outs, should result in a single ngModelChange event.

## Minimal reproduction of the problem with instructions
`<input [ngModel]="address.placeName" (ngModelChange)="placeNameUpdated($event)" [ngModelOptions]="{updateOn: 'blur'}">`

## What is the motivation / use case for changing the behavior?
Firing the ngModelChange on blur regardless of whether the model changed or not results in false positives.

## Environment

<pre><code>
Angular version: 5.0.0

Browser:
- [x] Chrome (desktop) version  61.0.3163.100
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: v8.9.0
- Platform:  Windows

Others:

</code></pre>

### PR DESCRIPTION
**Pull request by kara**
Fixes #20259.

### CODE REVIEW

File: packages/forms/src/directives/shared.ts [hunk 1]
```diff
@@ -171,7 +173,7 @@ export function syncPendingControls(form: FormGroup, directives: NgControl[]): v
   form._syncPendingControls();
   directives.forEach(dir => {
     const control = dir.control as FormControl;
-    if (control.updateOn === 'submit') {
+    if (control.updateOn === 'submit' && control._pendingChange) {
       dir.viewToModelUpdate(control._pendingValue);
```
Comment (tinayuangao) [inline packages/forms/src/directives/shared.ts]: no need to set `_pendingChange` to false after this?

**Review by tinayuangao — COMMENTED** [review]
Also reset `_pendingChange` when reset?

File: packages/forms/src/directives/shared.ts (hunk 1 above)
Comment (kara) [reply packages/forms/src/directives/shared.ts]: Ack, you're right! Should be fixed.

**Review by tinayuangao — APPROVED** [review]
LGTM

**Pull request closed without merging**